sdk/iot/core/doc/mqtt_state_machine.md

```diff
@@ -136,3 +138,120 @@ _Example:_
+
+The following APIs may be used to determine if the status indicates an error and if the operation should be retried:
+
+```C
+az_iot_hub_client_status status = response.status;
+if (az_iot_hub_client_is_success_status(status))
+{
+    // success case 
+}
+else
+{
+    if (az_iot_hub_client_is_retriable_status(status))
+    {
+        // retry
+    }
+    else
+    {
+        // fail
+    }
+}
+```
+
+##### Provisioning Service Errors
+
+Each Provisioning operation will return a status encoded as `az_iot_provisioning_client_status`.
+After the register operation is complete, the `az_iot_provisioning_client_registration_state registration_information` field will contain the `az_iot_provisioning_client_status error_code` that represents the overall status of the registration operation.  
+
+The following APIs may be used to determine if the status indicates error and if the error should be retried:
+
+```C
+az_iot_provisioning_client_status status = result.registration_information.error_code;
+if (az_iot_provisioning_client_is_success_status(status))
+{
+    // success case 
+}
+else
+{
+    if (az_iot_provisioning_client_is_retriable_status(status))
+    {
+        // retry
+    }
+    else
+    {
+        // fail
+    }
+}
+```
+
+#### Retry timing
+
+Network timeouts and the MQTT keep-alive interval should be configured considering tradeoffs between how fast network issues are detected vs traffic overheads. [This document](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-mqtt-support#default-keep-alive-timeout) describes the recommended keep-alive timeouts as well as the minimum idle timeout supported by Azure IoT services.
+
+For connectivity issues at all layers (TCP, TLS, MQTT) as well as cases where there is no `retry-after` sent by the service, we suggest using an exponential back-off with random jitter function. `_az_retry_calc_delay` is available in Azure SDK Core:
+
+```C
+// random_ms is a random value between 0 and max_jitter_msec
+// The previous operation took operation_msec
+
+int32_t total_delay_msec =  random_msec + _az_retry_calc_delay(attempt, min_retry_delay_msec, max_retry_delay_msec);
+int32_t delay_msec = max(operation_msec - total_delay_msec, 0); 
+```
+
+_Note 1_: The network stack may have used more time than the recommended delay before timing out. (e.g. The operation timed out after 2 minutes while the delay between operations is 1 second). In this case there is no need to delay the next operation.
+
+_Note 2_: To determine the parameters of the exponential with back-off retry strategy, we recommend modeling the network characteristics (including failure-modes). Compare the results with defined SLAs for device connectivity (e.g. 1M devices must be connected in under 30 minutes) and with the available IoT Azure scale (especially consider _throttling_, _quotas_ and maximum _requests/connects per second_).
+
+In the absence of modeling, we recommend the following default:

```

Comment:
Nit: the term modeling is used all over the place in PnP.  I get "network modeling" is also a thing, but if we could not use "monitoring" here maybe better.  

Also I think it should be "defaults" and not "default" since it's per value.

Maybe something like "We recommend the following defaults unless your network has more specific requirements."

Reply:
Model used in PnP is very different from the context in which modeling is used here. (https://en.wikipedia.org/wiki/Modeling_and_simulation)

See the disambiguation here https://en.wikipedia.org/wiki/Conceptual_model_(computer_science):
> Conceptual modeling should not be confused with other modeling disciplines such as data modelling, logical modelling and physical modelling. 


I don't agree with the reasoning that if PnP uses the noun form we can't use the verb here.